elements/lisk-bft/src/bft.ts

```diff
@@ -233,14 +232,11 @@ export class BFT extends EventEmitter {
 						storedFinalizedHeightBuffer,
 				  ).finalizedHeight;
 
-		/* Check BFT migration height
-		 https://github.com/LiskHQ/lips/blob/master/proposals/lip-0014.md#backwards-compatibility */
-		const bftMigrationHeight =
-			this.constants.startingHeight -
-			this.constants.activeDelegates * BFT_MIGRATION_ROUND_OFFSET;
-
-		// Choose max between stored finalized height or migration height
-		const finalizedHeight = Math.max(finalizedHeightStored, bftMigrationHeight);
+		// Choose max between stored finalized height or genesis height
+		const finalizedHeight = Math.max(
```

Comment:
Initially, the finalized height is the height of the genesis block. Why are we taking the maximum here?

Reply:
when the application restarts, it takes from the stored value and compare if it should be genesis height or the stored one. In most of the cases it should be the stored value.
ie)
1. running fresh it will choose genesis height
2. stop and run again, it will choose the genesis height

but maybe i can just initialize with genesis height if not exist

Reply:
But the genesis block should be processed only once, right? This block cannot be reverted and the finalized height can only be increased after processing blocks building on the genesis block.

Reply:
yes, so once the finalized height exist in storage, it will never go back. 
I will change it to to initialize when not existing in the storage

Reply:
Okay. Can be resolved then.